### PR TITLE
Update RPC Responses For V0.12.1

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Update to v0.12.1

**What does this PR do? Why is it needed?**

- [x] Updates error responses to be of a bounded list, instead of a previously unbounded 
list.
- [x] Updates block roots request to be of a bounded list also,so that peers do not request 
   a large amount of roots at once.
- [x] Makes sure chunked responses adhere to the max request block limit.
- [x] Fix outdated references.

**Which issues(s) does this PR fix?**

Part of #6076 

**Other notes for review**

Follows on from https://github.com/ethereum/eth2.0-specs/pull/1837 in the spec.

### CODE REVIEW

File: beacon-chain/sync/error.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 	"errors"
 	"io"
 
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+
```
Comment (nisdas) [inline beacon-chain/sync/error.go:9]:
```suggestion
```

File: beacon-chain/sync/rpc_beacon_blocks_by_root.go [hunk 2]
```diff
@@ -5,6 +5,10 @@ import (
 	"io"
 	"time"
 
+	pbp2p "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+
```
Comment (nisdas) [inline beacon-chain/sync/rpc_beacon_blocks_by_root.go:11]:
```suggestion
```

File: beacon-chain/sync/rpc_beacon_blocks_by_root_test.go [hunk 3]
```diff
@@ -7,6 +7,8 @@ import (
 	"testing"
 	"time"
 
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+
```
Comment (nisdas) [inline beacon-chain/sync/rpc_beacon_blocks_by_root_test.go:11]:
```suggestion
```

File: beacon-chain/sync/rpc_beacon_blocks_by_root.go [hunk 4]
```diff
@@ -99,10 +107,22 @@ func (r *Service) beaconBlocksRootRPCHandler(ctx context.Context, msg interface{
...
 	}
 
-	r.blocksRateLimiter.Add(stream.Conn().RemotePeer().String(), int64(len(blockRoots)))
+	if uint64(len(req.BlockRoots)) > params.BeaconNetworkConfig().MaxRequestBlocks {
+		resp, err := r.generateErrorResponse(responseCodeInvalidRequest, "requested more than the max block limit")
+		if err != nil {
+			log.WithError(err).Error("Failed to generate a response error")
+		} else {
+			if _, err := stream.Write(resp); err != nil {
+				log.WithError(err).Errorf("Failed to write to stream")
```
Comment (rauljordan) [inline beacon-chain/sync/rpc_beacon_blocks_by_root.go:116]: Should we return in these conditions?

Comment (nisdas) [reply beacon-chain/sync/rpc_beacon_blocks_by_root.go:116]: we do return the error  below it

**Pull request merged**